src/cpp/flann/algorithms/kmeans_index.h

```diff
@@ -1019,8 +1033,19 @@ class KMeansIndex : public NNIndex<Distance>
     	std::swap(chooseCenters_, other.chooseCenters_);
     }
 
+    void prepSearchHeap()
+    {
+    	heaps = new std::vector<Heap<BranchSt>*>(omp_get_max_threads());
```

Comment:
`omp_get_max_threads` isn't guaranteed to be big enough if the user requests a specific number of threads ([docs](https://msdn.microsoft.com/en-us/library/ewb30w8w.aspx)). I think the accesses above need to check if `heap` needs to be resized....